Error. compiler crashed (exit code 4) - cannot access unnamed package

**Failure link**
---
From internal [Test_openjdk21_j9_extended.openjdk_x86-64_linux_testList_0](https://hyc-runtimes-jenkins.swg-devops.com/job/Test_openjdk21_j9_extended.openjdk_x86-64_linux_testList_0/139/) (ubu20x86-svl-rt2-1)

```**Java Version**
java version "21.0.6-beta" 2025-01-21
IBM Semeru Runtime Certified Edition 21.0.6+6-202412071603 (build 21.0.6-beta+6-202412071603)
Eclipse OpenJ9 VM 21.0.6+6-202412071603 (build master-501751012f, JRE 21 Linux amd64-64-Bit Compressed References 20241207_325 (JIT enabled, AOT enabled)
OpenJ9   - 501751012f
OMR      - b893b2be2
JCL      - 71fff4eae based on jdk-21.0.6+6)
```


[Rerun in Grinder](https://hyc-runtimes-jenkins.swg-devops.com/job/Grinder/parambuild/?SDK_RESOURCE=customized&amp;TARGET=jdk_vector_1&amp;BASE_DOCKER_REGISTRY_CREDENTIAL_ID=&amp;TEST_FLAG=&amp;UPSTREAM_TEST_JOB_NAME=Test_openjdk21_j9_extended.openjdk_x86-64_linux&amp;DOCKER_REQUIRED=false&amp;ACTIVE_NODE_TIMEOUT=5&amp;VENDOR_TEST_DIRS=&amp;EXTRA_DOCKER_ARGS=&amp;TKG_OWNER_BRANCH=adoptium%3Amaster&amp;OPENJ9_SYSTEMTEST_OWNER_BRANCH=eclipse%3Amaster&amp;PLATFORM=x86-64_linux&amp;GENERATE_JOBS=false&amp;KEEP_REPORTDIR=true&amp;PERSONAL_BUILD=false&amp;DOCKER_REGISTRY_DIR=&amp;RERUN_ITERATIONS=0&amp;ADOPTOPENJDK_REPO=https%3A%2F%2Fgithub.com%2Fadoptium%2Faqa-tests.git&amp;SETUP_JCK_RUN=false&amp;DOCKER_REGISTRY_URL_CREDENTIAL_ID=&amp;LABEL=&amp;EXTRA_OPTIONS=&amp;CUSTOMIZED_SDK_URL=https%3A%2F%2Fna-public.artifactory.swg-devops.com%2Fartifactory%2Fsys-rt-generic-local%2Fhyc-runtimes-jenkins.swg-devops.com%2Fbuild-scripts%2Fjobs%2Fjdk21u%2Fjdk21u-linux-x64-openj9-IBM%2F325%2Fibm-semeru-certified-jdk_x64_linux_JDK21U_2024-12-07-16-01.tar.gz+https%3A%2F%2Fna-public.artifactory.swg-devops.com%2Fartifactory%2Fsys-rt-generic-local%2Fhyc-runtimes-jenkins.swg-devops.com%2Fbuild-scripts%2Fjobs%2Fjdk21u%2Fjdk21u-linux-x64-openj9-IBM%2F325%2Fibm-semeru-certified-testimage_x64_linux_JDK21U_2024-12-07-16-01.tar.gz&amp;BUILD_IDENTIFIER=&amp;JENKINS_KEY=Jenkins+4096+key&amp;ADOPTOPENJDK_BRANCH=master&amp;LIGHT_WEIGHT_CHECKOUT=false&amp;USE_JRE=false&amp;ARTIFACTORY_SERVER=na.artifactory.swg-devops&amp;KEEP_WORKSPACE=false&amp;USER_CREDENTIALS_ID=83181e25-eea4-4f55-8b3e-e79615733226&amp;JDK_VERSION=21&amp;DOCKER_REGISTRY_URL=&amp;ITERATIONS=1&amp;VENDOR_TEST_REPOS=&amp;JDK_REPO=git%40github.com%3Aibmruntimes%2Fopenj9-openjdk-jdk21&amp;JCK_GIT_BRANCH=main&amp;RELEASE_TAG=&amp;OPENJ9_BRANCH=master&amp;OPENJ9_SHA=&amp;JCK_GIT_REPO=&amp;VENDOR_TEST_BRANCHES=&amp;OPENJ9_REPO=https%3A%2F%2Fgithub.com%2Feclipse-openj9%2Fopenj9.git&amp;UPSTREAM_JOB_NAME=&amp;CLOUD_PROVIDER=&amp;CUSTOM_TARGET=&amp;VENDOR_TEST_SHAS=&amp;JDK_BRANCH=openj9&amp;LABEL_ADDITION=%21%28sw.os.cent.6%7C%7Csw.os.rhel.6%29&amp;ARTIFACTORY_REPO=sys-rt-generic-local%2Chyc-runtimes-team-restricted-oracle-sdk-generic-local&amp;ARTIFACTORY_ROOT_DIR=&amp;UPSTREAM_TEST_JOB_NUMBER=176&amp;DOCKERIMAGE_TAG=&amp;TEST_TIME=&amp;JDK_IMPL=openj9&amp;SSH_AGENT_CREDENTIAL=83181e25-eea4-4f55-8b3e-e79615733226&amp;AUTO_DETECT=true&amp;SLACK_CHANNEL=%23rt-jenkins&amp;DYNAMIC_COMPILE=false&amp;RELATED_NODES=&amp;ADOPTOPENJDK_SYSTEMTEST_OWNER_BRANCH=adoptium%3Amaster&amp;APPLICATION_OPTIONS=&amp;CUSTOMIZED_SDK_URL_CREDENTIAL_ID=7c1c2c28-650f-49e0-afd1-ca6b60479546&amp;ARCHIVE_TEST_RESULTS=false&amp;NUM_MACHINES=&amp;OPENJDK_SHA=&amp;TRSS_URL=https%3A%2F%2Ftrssrtp1.fyre.ibm.com%2F&amp;USE_TESTENV_PROPERTIES=false&amp;BUILD_LIST=openjdk&amp;ADDITIONAL_ARTIFACTS_REQUIRED=&amp;UPSTREAM_JOB_NUMBER=&amp;STF_OWNER_BRANCH=adoptium%3Amaster&amp;TIME_LIMIT=25&amp;JVM_OPTIONS=&amp;PARALLEL=None) - Change TARGET to run only the failed test targets

**Optional info**
---
Failure output (captured from console output)
---
```
[2024-12-07T18:58:32.648Z] variation: Mode650
[2024-12-07T18:58:32.648Z] JVM_OPTIONS:  -XX:-UseCompressedOops -Xverbosegclog 

[2024-12-07T18:58:53.178Z] TEST: jdk/incubator/vector/Byte256VectorLoadStoreTests.java

[2024-12-07T18:58:53.180Z] warning: using incubating module(s): jdk.incubator.vector
[2024-12-07T18:58:53.180Z] /home/jenkins/workspace/Test_openjdk21_j9_extended.openjdk_x86-64_linux_testList_0/aqa-tests/openjdk/openjdk-jdk/test/jdk/jdk/incubator/vector/Byte256VectorLoadStoreTests.java:34: error: cannot access unnamed package
[2024-12-07T18:58:53.180Z] import java.lang.foreign.MemorySegment;
[2024-12-07T18:58:53.180Z] ^
[2024-12-07T18:58:53.180Z]   /org/testng/internal/annotations/DefaultAnnotationTransformer.class
[2024-12-07T18:58:53.180Z] 1 error
[2024-12-07T18:58:53.180Z] 1 warning
[2024-12-07T18:58:53.180Z] printing javac parameters to: /home/jenkins/workspace/Test_openjdk21_j9_extended.openjdk_x86-64_linux_testList_0/aqa-tests/TKG/output_17335904872610/jdk_vector_1/work/scratch/2/javac.20241207_105852.args
[2024-12-07T18:58:53.180Z] 
[2024-12-07T18:58:53.180Z] TEST RESULT: Error. compiler crashed (exit code 4)

[2024-12-07T19:23:24.065Z] jdk_vector_1_FAILED
```
javac.20241207_105852.args
```
--add-modules
jdk.incubator.vector,java.base
--add-exports
java.base/jdk.internal.vm.annotation=ALL-UNNAMED
-d
/home/jenkins/workspace/Test_openjdk21_j9_extended.openjdk_x86-64_linux_testList_0/aqa-tests/TKG/output_17335904872610/jdk_vector_1/work/classes/0/jdk/incubator/vector/Byte256VectorLoadStoreTests.d
-sourcepath
/home/jenkins/workspace/Test_openjdk21_j9_extended.openjdk_x86-64_linux_testList_0/aqa-tests/openjdk/openjdk-jdk/test/jdk/jdk/incubator/vector
-classpath
/home/jenkins/workspace/Test_openjdk21_j9_extended.openjdk_x86-64_linux_testList_0/aqa-tests/openjdk/openjdk-jdk/test/jdk/jdk/incubator/vector:/home/jenkins/workspace/Test_openjdk21_j9_extended.openjdk_x86-64_linux_testList_0/aqa-tests/TKG/output_17335904872610/jdk_vector_1/work/classes/0/jdk/incubator/vector/Byte256VectorLoadStoreTests.d:/home/jenkins/workspace/Test_openjdk21_j9_extended.openjdk_x86-64_linux_testList_0/jvmtest/openjdk/jtreg/lib/testng-7.3.0.jar:/home/jenkins/workspace/Test_openjdk21_j9_extended.openjdk_x86-64_linux_testList_0/jvmtest/openjdk/jtreg/lib/guice-5.1.0.jar:/home/jenkins/workspace/Test_openjdk21_j9_extended.openjdk_x86-64_linux_testList_0/jvmtest/openjdk/jtreg/lib/jcommander-1.82.jar
--enable-preview
--source=21
/home/jenkins/workspace/Test_openjdk21_j9_extended.openjdk_x86-64_linux_testList_0/aqa-tests/openjdk/openjdk-jdk/test/jdk/jdk/incubator/vector/Byte256VectorLoadStoreTests.java
```

[50x internal Grinder](https://hyc-runtimes-jenkins.swg-devops.com/job/Grinder/45760/) - failed at `rhel8x86-svl-rt1-1` and passed in the other 3 machines (rhel9x86-rt-2, rtj-rhel8x86-rtp-test-xemvx-1, cent9x86-rtp-rt1-1)


[openjdk21_j9_extended.openjdk_x86-64_linux jdk_security3_1_FAILED](https://hyc-runtimes-jenkins.swg-devops.com/job/Test_openjdk21_j9_extended.openjdk_x86-64_linux_testList_0/139/)
```
[2024-12-07T18:01:05.476Z] TEST: javax/security/auth/PrivateCredentialPermission/MoreThenOnePrincipals.java

[2024-12-07T18:01:05.477Z] /home/jenkins/workspace/Test_openjdk21_j9_extended.openjdk_x86-64_linux_testList_0/aqa-tests/openjdk/openjdk-jdk/test/jdk/javax/security/auth/PrivateCredentialPermission/MoreThenOnePrincipals.java:24: error: cannot access unnamed package
[2024-12-07T18:01:05.477Z] import com.sun.security.auth.NTUserPrincipal;
[2024-12-07T18:01:05.477Z] ^
[2024-12-07T18:01:05.477Z]   /org/testng/internal/annotations/DefaultAnnotationTransformer.class
[2024-12-07T18:01:05.477Z] 1 error
[2024-12-07T18:01:05.477Z] printing javac parameters to: /home/jenkins/workspace/Test_openjdk21_j9_extended.openjdk_x86-64_linux_testList_0/aqa-tests/TKG/output_17335904872610/jdk_security3_1/work/scratch/1/javac.20241207_100055.args
[2024-12-07T18:01:05.477Z] 
[2024-12-07T18:01:05.477Z] TEST RESULT: Error. compiler crashed (exit code 4)
```
javac.20241207_100055.args
```
--add-modules
jdk.security.auth
-d
/home/jenkins/workspace/Test_openjdk21_j9_extended.openjdk_x86-64_linux_testList_0/aqa-tests/TKG/output_17335904872610/jdk_security3_1/work/classes/2/javax/security/auth/PrivateCredentialPermission/MoreThenOnePrincipals.d
-sourcepath
/home/jenkins/workspace/Test_openjdk21_j9_extended.openjdk_x86-64_linux_testList_0/aqa-tests/openjdk/openjdk-jdk/test/jdk/javax/security/auth/PrivateCredentialPermission
-classpath
/home/jenkins/workspace/Test_openjdk21_j9_extended.openjdk_x86-64_linux_testList_0/aqa-tests/openjdk/openjdk-jdk/test/jdk/javax/security/auth/PrivateCredentialPermission:/home/jenkins/workspace/Test_openjdk21_j9_extended.openjdk_x86-64_linux_testList_0/aqa-tests/TKG/output_17335904872610/jdk_security3_1/work/classes/2/javax/security/auth/PrivateCredentialPermission/MoreThenOnePrincipals.d:/home/jenkins/workspace/Test_openjdk21_j9_extended.openjdk_x86-64_linux_testList_0/jvmtest/openjdk/jtreg/lib/testng-7.3.0.jar:/home/jenkins/workspace/Test_openjdk21_j9_extended.openjdk_x86-64_linux_testList_0/jvmtest/openjdk/jtreg/lib/guice-5.1.0.jar:/home/jenkins/workspace/Test_openjdk21_j9_extended.openjdk_x86-64_linux_testList_0/jvmtest/openjdk/jtreg/lib/jcommander-1.82.jar
/home/jenkins/workspace/Test_openjdk21_j9_extended.openjdk_x86-64_linux_testList_0/aqa-tests/openjdk/openjdk-jdk/test/jdk/javax/security/auth/PrivateCredentialPermission/MoreThenOnePrincipals.java
```

--- a/runtime/compiler/x/codegen/J9TreeEvaluator.cpp
+++ b/runtime/compiler/x/codegen/J9TreeEvaluator.cpp
@@ -9656,7 +9656,7 @@ J9::X86::TreeEvaluator::vectorizedHashCodeHelper(TR::Node *node, TR::DataType dt
    TR::LabelSymbol *residueLoopLabel = generateLabelSymbol(cg);
 
    residueBeginLoopLabel->setStartInternalControlFlow();
-   residueBeginLoopLabel->setEndInternalControlFlow();
+   residueEndLoopLabel->setEndInternalControlFlow();
 
    generateLabelInstruction(TR::InstOpCode::label, node, residueBeginLoopLabel, cg);
    generateRegRegInstruction(TR::InstOpCode::CMP4RegReg, node, index, length, cg);

